docs/specification.md

```diff
@@ -1,153 +1,207 @@
 
-- result: [`ExecutionPayloadHeaderV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/bellatrix/beacon-chain.md#executionpayloadheader)
-- error: code and message set in case an exception happens while getting the payload.
+| Code | Message | Meaning |
+| - | - | - |
+| -32000 | Server error | Generic client error while processing request. |
+| -32001 | Unknown hash | No block with the provided hash is known. |
+| -32002 | Unknown validator | No known mapping between validator and feeRecipient. |
+| -32003 | Invalid SSZ | Unable to decode SSZ. |
+| -32004 | Unknown block | Block does not match the provided header. |
+| -32005 | Invalid signature | Provided signature is invalid. |
+| -32006 | Invalid timestamp | Provided timestamp was invalid. |
+| -32600 | Invalid request | The JSON sent is not a valid Request object. |
+| -32601 | Method not found | The method does not exist / is not available. |
+| -32602 | Invalid params | Invalid method parameter(s). |
+| -32603 | Internal error | Internal JSON-RPC error. |
+| -32700 | Parse error | Invalid JSON was received by the server. |
 
-### relay_getPayloadHeaderV1
+## Routines
 
-#### Request
+### Signing
 
-- method: `relay_getPayloadHeaderV1`
-- params:
-  1. `payloadId`: `DATA`, 8 Bytes - Identifier of the payload build process
+All signature operations should follow the [standard BLS operations][bls] interface defined in `consensus-specs`.
 
-#### Response
+There are two types of data to sign over in the Builder API:
+* In-protocol messages, e.g. [`BlindBeaconBlock`](#blindbeaconblock), which should compute the signing root using [`computer_signing_root`][compute-signing-root] and use the domain specified for beacon block proposals.
+* Builder API messages, e.g. [`builder_setFeeRecipientV1`](#builder_setFeeRecipientV1), which should compute the signing root using [`compute_signing_root`][compute-signing-root] and the domain `DomainType('0xXXXXXXXX')` (TODO: get a proper domain).
 
-- result: [`SignedMEVPayloadHeader`](#signedmevpayloadheader)
-- error: code and message set in case an exception happens while getting the payload.
+As `compute_signing_root` takes `SSZObject` as input, client software should convert in-protocol messages to their SSZ representation to compute the signing root and Builder API messages to the SSZ representations defined [above](#sszobjects).
+
+## Methods
 
-### relay_proposeBlindedBlockV1
+### `builder_setFeeRecipientV1`
 
 #### Request
 
-- method: `relay_proposeBlindedBlockV1`
+- method: `builder_setFeeRecipientV1`
 - params:
-  1. [`SignedBlindedBeaconBlock`](#signedblindedbeaconblock)
-  2. [`SignedMEVPayloadHeader`](#signedmevpayloadheader)
+  1. `feeRecipient`: `DATA`, 20 Bytes - Address of account which should receive fees.
+  2. `timestamp`: `QUANTITY`, uint64 - Unix timestamp of announcement.
+  2. `publicKey`: `DATA`, 48 Bytes - Public key of validator.
+  3. `signature`: `DATA`, 96 Bytes - Signature over `feeRecipient`.
```

Comment:
```suggestion
  3. `publicKey`: `DATA`, 48 Bytes - Public key of validator.
  4. `signature`: `DATA`, 96 Bytes - Signature over `feeRecipient`.
```

Reply:
> 4. `signature`: `DATA`, 96 Bytes - Signature over `feeRecipient`.

Should this be "Signature over `feeRecipient + timestamp`?

Reply:
Yes, thank you for pointing out. I will update.